### PR TITLE
Fixes #6353: Secure agent/server communication

### PR DESCRIPTION
**Pull request by ncharles**
https://www.rudder-project.org/redmine/issues/6363

### CODE REVIEW

**Comment by peckpeck** [comment]
I don't like this pull request being in the issue meant to aggregate all security related improvement.
This hould be in https://www.rudder-project.org/redmine/issues/6353 instead.

File: techniques/system/common/1.0/failsafe.st [hunk 1]
```diff
@@ -27,7 +27,7 @@ body common control
         inputs             => { "common/1.0/update.cf", "common/1.0/rudder-stdlib-core.cf" };
         output_prefix      => "rudder";
 
-        protocol_version   => "classic";
+        protocol_version   => "2";
```
Comment (peckpeck) [inline techniques/system/common/1.0/failsafe.st]: This won't work on older servers. We need a transition method.
- easy path: leave classic until old server are not supported anymore
- bad path: ask the user
- hard path: test both cases starting with the secure one allowing an override to forbid the old one

Comment (peckpeck) [inline techniques/system/common/1.0/failsafe.st]: Alternative: 
- old protocol in initial promises
- new protocol in system technique

**Comment by ncharles** [comment]
ha, i'm retargeting the PR, sorry (it's a typo)

**Comment by ncharles** [comment]
commit renamed

File: techniques/system/common/1.0/failsafe.st (hunk 1 above)
Comment (ncharles) [inline techniques/system/common/1.0/failsafe.st]: An agent protocol v2 can connect to a server protocol v1 (i just tried it)

**Pull request merged**